core/client/db/src/cache/mod.rs

```diff
@@ -159,16 +237,24 @@ impl<'a, Block: BlockT> DbCacheTransaction<'a, Block> {
 		parent: ComplexBlockId<Block>,
 		block: ComplexBlockId<Block>
 	) -> ClientResult<Self> {
-		assert!(self.authorities_at_op.is_none());
+		assert!(self.cache_at_op.is_empty());
 
-		self.authorities_at_op = self.cache.authorities_at.on_block_finalize(
-			&mut self::list_storage::DbStorageTransaction::new(
-				self.cache.authorities_at.storage(),
-				&mut self.tx
-			),
-			parent,
-			block,
-		)?;
+		for (name, cache_at) in self.cache.cache_at.iter() {
+			let op = cache_at.on_block_finalize(
+				&mut self::list_storage::DbStorageTransaction::new(
+					cache_at.storage(),
+					&mut self.tx
+				),
+				parent.clone(),
+				block.clone(),
+			)?;
+
+			if let Some(op) = op {
+				self.cache_at_op.insert(name.to_owned(), op);
+			}
+		}
+
+		self.best_finalized_block = Some(block.clone());
```

Comment:
```suggestion
		self.best_finalized_block = Some(block);
```

Should work as well?